src/main/java/uk/gov/companieshouse/api/testdata/service/impl/UserServiceImpl.java

```diff
@@ -82,21 +163,66 @@ List<String> processRoles(List<String> roles) throws DataException {
     }
 
     UserRoles getUserRole(String roleName) throws DataException {
+        LOG.debug("getUserRole called with roleName= " + roleName);
         if (roleName == null) {
+            LOG.error("Invalid role name: null");
             throw new DataException("Invalid role name: null");
         }
         try {
-            return UserRoles.valueOf(roleName);
+            var role = UserRoles.valueOf(roleName);
+            LOG.debug("Found UserRoles enum for roleName= " + roleName);
+            return role;
         } catch (IllegalArgumentException error) {
-            throw new DataException("Invalid role name: " + roleName);
+            LOG.error("Invalid role name provided: "
+                    + roleName);
+            throw new DataException("Invalid role name: "
+                    + roleName);
         }
     }
 
     @Override
+    @Transactional
     public boolean delete(String userId) {
-        var user = repository.findById(userId);
-        user.ifPresent(repository::delete);
-        return user.isPresent();
+        LOG.info("delete called for userId= " + userId);
+        var userOpt = repository.findById(userId);
+        if (userOpt.isEmpty()) {
+            LOG.debug("User not found for id=  " + userId);
+            return false;
+        }
+
+        Optional<Identity> identityOpt = identityRepository.findByUserId(userId);
+        if (identityOpt.isPresent()) {
+            var identity = identityOpt.get();
+            LOG.info("Found identity for userId = "
+                    + userId
+                    + "Identity_id = " + identity.getId());
+            try {
+                uvidRepository.deleteByIdentityId(identity.getId());
+                LOG.debug("Deleted UVIDs for identityId = " + identity.getId());
+            } catch (Exception er) {
+                LOG.info("Failed to delete UVIDs for identityId = "
```

Comment:
errors should be log.Error - shouldn't we throw an exception here as we have had a failure to delete not just allow it to carry on?